contracts/modules/staking/StakingProducts.sol

```diff
@@ -162,12 +164,33 @@ contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
       }
 
       if (_param.setTargetPrice) {
+
         if (_param.targetPrice > TARGET_PRICE_DENOMINATOR) {
           revert TargetPriceTooHigh();
         }
+
         if (_param.targetPrice < globalMinPriceRatio) {
           revert TargetPriceBelowMin();
         }
+
+        // if this is an existing product, when the target price is updated we need to calculate the
+        // current base price using the old target price and update the bumped price to that value
+        // uses the same logic as calculatePremium()
+        if (!isNewProduct) {
```

Comment:
how about we only do this if the targetPrice is different?

Seems like with current code path you can have setTargetPrice = true and put the same targetPrice

Reply:
In that case you just don't need `setTargetPrice = true`, no?